### PR TITLE
Add regression test for wasm bug

### LINKED ISSUE #76281: rustc hangs when compiling for wasm32
### hang.rs

```Rust
fn main() {
    let mut v: Vec<&()> = Vec::new();
    v.sort_by_key(|&r| r as *const ());
}
```

### Meta

`rustc --version --verbose`:
```
rustc --version --verbose
rustc 1.46.0 (04488afe3 2020-08-24)
binary: rustc
commit-hash: 04488afe34512aa4c33566eb16d8c912a3ae04f9
commit-date: 2020-08-24
host: x86_64-apple-darwin
release: 1.46.0
LLVM version: 10.0
```

### Error output
I compile with this:

```
rustc hang.rs -C opt-level=2 --target wasm32-unknown-unknown
```

And that hangs `rustc`.

### Details

`opt-level` must be 2 or higher, and `--target` must be `wasm32-unknown-unknown`.

It does NOT hang with `rustc 1.48.0-nightly (d006f5734 2020-08-28)`, so possibly this is a bug that has already been fixed.

### PR DESCRIPTION
**Pull request by rossmacarthur**
Fixes #76281

### CODE REVIEW

File: src/test/ui/issues/issue-76281.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// only-wasm32
+// compile-flags: -C opt-level=2 --target wasm32-unknown-unknown
```
Comment (bjorn3) [inline src/test/ui/issues/issue-76281.rs:2]:
```suggestion
// compile-flags: -C opt-level=2
```

`only-wasm32` implies that `--target wasm32-unknown-unknown` is used anyway. This test could also be useful on other targets, so `only-wasm32` should also be removed I think.

**Comment by Mark-Simulacrum** [comment]
Hm, do we have a bisection here? I hesitate to add a test without knowing what PR fixed it.

It looks like you need to add `// run-pass` to the test to get CI to pass.

**Comment by camelid** [comment]
@rossmacarthur Could you address the review comments?

**Comment by JohnCSimon** [comment]
Pinging from triage: @rossmacarthur 
Unfortunately this has been idle for too long and I'm going to close it as inactive, please feel free to reopen when you get the chance to address the build issues and merge conflicts.

**Pull request closed without merging**